lib/core/src/sdk.rs

```diff
@@ -1180,16 +1348,71 @@ impl LiquidSdk {
     /// # Returns
     ///
     /// * A [ReceivePaymentResponse] containing:
-    ///     * `invoice` - the bolt11 Lightning invoice that should be paid
+    ///     * `destination` - the final destination to be paid by the payer, either a BIP21 URI (Liquid or Bitcoin), a Liquid address or an invoice
     pub async fn receive_payment(
         &self,
         req: &ReceivePaymentRequest,
     ) -> Result<ReceivePaymentResponse, PaymentError> {
         self.ensure_is_started().await?;
 
-        let payer_amount_sat = req.prepare_res.payer_amount_sat;
-        let fees_sat = req.prepare_res.fees_sat;
+        let PrepareReceiveResponse {
+            payment_method,
+            amount_sat,
+            fees_sat,
+        } = &req.prepare_response;
 
+        match payment_method {
+            PaymentMethod::Lightning => {
+                let Some(amount_sat) = amount_sat else {
+                    return Err(PaymentError::AmountMissing { err: "`amount_sat` must be specified when `PaymentMethod::Lightning` is used.".to_string() });
+                };
+                self.create_receive_swap(*amount_sat, *fees_sat, req.description.clone())
+                    .await
+            }
+            PaymentMethod::BitcoinAddress => {
+                let Some(amount_sat) = amount_sat else {
+                    return Err(PaymentError::AmountMissing { err: "`amount_sat` must be specified when `PaymentMethod::BitcoinAddress` is used.".to_string() });
+                };
+                self.receive_onchain(*amount_sat, *fees_sat).await
+            }
+            PaymentMethod::LiquidAddress => {
+                let address = self.onchain_wallet.next_unused_address().await?.to_string();
+
+                let receive_destination = match amount_sat {
+                    Some(amount_sat) => LiquidAddressData {
+                        address: address.to_string(),
+                        network: self.config.network.into(),
+                        amount_sat: Some(*amount_sat),
+                        asset_id: Some(match self.config.network {
+                            LiquidNetwork::Mainnet => AssetId::LIQUID_BTC.to_hex(),
+                            LiquidNetwork::Testnet => {
+                                "6f0279e9ed041c3d710a9f57d0c02928416460c4b722ae3457a11eec381c526d"
```

Comment:
This is the same as `AssetId::LIQUID_BTC` :
https://github.com/ElementsProject/rust-elements/blob/f875ae2ac247a44ca1bbbc4aa16a3c7adbad6c4d/src/issuance.rs#L367-L370

This block should be replaced with
```rust
asset_id: Some(AssetId::LIQUID_BTC.to_hex()),
```